src/Controllers/HistoryControllerFactory.php

```diff
@@ -21,31 +23,30 @@ class HistoryControllerFactory implements Factory
     public function create($service, array $params = array())
     {
         // If no request is available yet, return the default controller
-        if (!Injector::inst()->has(HTTPRequest::class)) {
-            return new CMSPageHistoryController();
-        }
-        $request = Injector::inst()->get(HTTPRequest::class);
-        $id = $request->param('ID');
-
-        if ($id && is_numeric($id)) {
-            // Ensure we read from the draft stage at this position
-            $page = Versioned::get_one_by_stage(
-                SiteTree::class,
-                Versioned::DRAFT,
-                sprintf('"SiteTree"."ID" = \'%d\'', $id)
-            );
-
-            if ($page && $this->isEnabled($page)) {
-                return Injector::inst()->create(CMSPageHistoryViewerController::class);
+        if (Injector::inst()->has(HTTPRequest::class)) {
+            $request = Injector::inst()->get(HTTPRequest::class);
+            $id = $request->param('ID');
+
+            if ($id && is_numeric($id)) {
+                // Ensure we read from the draft stage at this position
+                $page = Versioned::get_one_by_stage(
+                    SiteTree::class,
+                    Versioned::DRAFT,
+                    sprintf('"SiteTree"."ID" = \'%d\'', $id)
+                );
+
+                if ($page && !$this->isEnabled($page)) {
+                    return new CMSPageHistoryController();
```

Comment:
Can you retain the comment about not using injector here?

Reply:
Done.